File: go/internal/sol/p0036/p0036.go
package p0036

import (
	"strconv"

	"pe-solver/internal/mylib"
)

func compute(limit int) string {
	var result int
	for i := 1; i < limit; i += 2 {
		if mylib.IsPalindrome(i, 10) && mylib.IsPalindrome(i, 2) {
			result += i
		}
	}

	return strconv.FormatInt(int64(result), 10)
}

func Solve() string {
	return compute(1_000_000)
}

File: haskell/src/Sol/P0036.hs
module Sol.P0036 (compute, solve) where

import Mylib.Math (isPalindrome)

compute :: Int -> String
compute limit =
    show
        . sum
        $ filter (\x -> isPalindrome x 10 && isPalindrome x 2) [1, 3 .. limit]

solve :: String
solve = compute 1_000_000
